### DISCUSSION TITLE
Ran C# Script, Auto-Rollback did not work

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1258] [🙋♀️TE3 Q&A] [answered]
I ran a C# script to create partitions.
It did create partitions.
I tried to Auto-Rollback, by pressing the auto-rollback button.  It did not rollback the partitions i just created. I had not committed the changes to the server.
This can't be correct.  Is this a bug?  Surely someone would have seen this before
I am manually having to undo

There is a little tooltip (confusing) that says ctrlz will undo.  It did NOT undo.  In fact all it did was ERASE the contents of the C# window!

**Comment by TinyTimothy2** [comment]
confusing to me, but perhaps not others. I have to move off the C# page and onto the table i created the partitions and then use ctrl z.

**Reply by otykier** [reply, marked as answer]
Everything you describe is by design. Auto-rollback is a toggle option. If enabled, Tabular Editor will undo the changes made by the script, *if the script fails*.

If the script completes successfully, you can undo it. However, CTRL+Z sends an “Undo” command to the view in focus. When the script itself is focused, that will just undo the last edit (e.g. if you just pasted some code in the script editor, the code will disappear). So you either need to change focus to the TOM Explorer, or another non-code editor view, or use the **Global undo** command, accessible through the **Edit** menu.
